cmake/AddPdfTarget.cmake

```diff
@@ -1,23 +1,46 @@
-function (add_pdf_target docname lang entities figures)
+#
+# A function to generate PDF documentation
+#
+# FUNCTION:
+#   add_pdf_target
+# ARGUMENTS:
+# - docname: The basename of the main xml file. Will be used to locate
+#            this primary xml file and for various output files/directories.
+#            Either "gnucash-guide" or "gnucash-help" now.
+# - lang: The language of the current document, such as "C", "de" and so on.
+# - entities: A list of all xml files this document is composed of. ONLY filename, WITHOUT path.
+#             It does NOT contain "${docname}.xml".
+# - figures: A list of FULL PATH image files.
+# - dtd_files: A list of FULL PATH DTD files.
+#
+#   Note: The figures and icons should be placed at ${CMAKE_CURRENT_BINARY_DIR}, not ${BUILD_DIR} for FOP.
+#
+function (add_pdf_target docname lang entities figures dtd_files)
 
+    # Prepare variables and directories for build
+    set(fmt "pdf")
     set(fofile "${docname}.fo")
+    set(outfile "${docname}.${fmt}")
+
+    set(BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/${fmt}")
+    set(OUTPUT_DIR "${CMAKE_BINARY_DIR}/share/doc/${lang}")
+
+    configure_file("${FOP_XCONF}" "${BUILD_DIR}/fop.xconf")
+
+    # GnuCash-specific xsl files
+    file(GLOB xsl_files "${CMAKE_SOURCE_DIR}/xsl/*.xsl")
+    file(GLOB gnucash_icon_files "${CMAKE_SOURCE_DIR}/xsl/icons/*")
+
     set(pdffile "${docname}.pdf")
 
     set(BUILD_DIR "${DOCDIR_BUILD}/${lang}")
-    file(MAKE_DIRECTORY "${BUILD_DIR}")
-
-    # Determine paper format depending on language (which maps to the document's directory name)
-    # * for language "C" (fallback language) determine paper format based on current locale
-    # * all others use A4.
-    set (XSLTFLAGS_FO "--stringparam paper.type A4")
-    if (lang STREQUAL "C")
-        # For the fallback language determine paper format depending on locale
-        # Only US or C will be set to letter. All others use A4.
-        set (ENV_LANG $ENV{LANG})
-        if (ENV_LANG AND ENV_LANG MATCHES ".*_us.*|C") # Replacing ENV_LANG here with if ($ENV{LANG}) won't work.
-            set (XSLTFLAGS_FO "--stringparam paper.type letter")
-        endif()
-    endif()
```

Comment:
Removing the page format resolution is a different topic from adding argument dtd_files. It should happen in a different commit. Ideally the commit that proposes another solution.